srcpkgs/i3status-rust/template

```diff
@@ -1,23 +1,24 @@
 # Template file for 'i3status-rust'
 pkgname=i3status-rust
-version=0.22.0
-revision=4
+version=0.32.3
+revision=1
 build_style=cargo
 make_check_args="--bins"
 hostmakedepends="pkg-config"
-makedepends="dbus-devel pulseaudio-devel openssl-devel libsensors-devel"
+makedepends="dbus-devel pulseaudio-devel openssl-devel libsensors-devel pandoc"
```

Comment:
pandoc can't be cross-built so this makes it impossible to build this for arm/aarch64

Reply:
Ok; With pandoc as a hostmakedepend, it fails the CI for other architectures (presumably because the cross-compiled cargo tries to call the different-architecture pandoc and then fails). I can't seem to find those CI runs, but I'm glad to change it back if you think that is the way to go.

Alternatively, since pandoc is now required to generate the man page for i3status-rust, do you have any other suggestions?

Reply:
see how ncspot does it

Reply:
maybe not using RUST_TARGET so that it builds for the host

Reply:
So after poking at this a little while, this seems to be complicated by the fact that i3status-rs generates its man page very differently from ncspot (it uses environment variables like CARGO_MANIFEST_DIR and explicitly walks the filesystem to build it, rather than ncspot's approach of just getting the args from the main file; this makes sense as it includes configuration documentation in various source files pertaining to the thing being configured).

Reply:
Part of the problem is that there are few diagnostics coming out of `vtargetrun`; whenever there is an error (other than an error with args for qemu) it just pops up:
```
=> i3status-rust-0.32.3_1: running post_build ...
    Finished release [optimized] target(s) in 0.09s
=> ERROR: i3status-rust-0.32.3_1: post_build: '"/usr/bin/qemu-${XBPS_TARGET_QEMU_MACHINE}-static" "$@"' exited with 1
=> ERROR:   in vtargetrun() at common/build-helper/qemu.sh:10
=> ERROR:   in post_build() at srcpkgs/i3status-rust/template:20
```
While I can manually poke the generated binary (and the command line I'm passing to it) for x86_64, the lack of diagnostics means that trying to make sure everything works for cross-compiling is a bit more challenging.

Reply:
Specifically, I've updated it to have this for a post_build (with just this and the addition of the build-helper, it is now failing on all builds...hence why I'm not pushing it):
```
post_build() {
	cargo auditable build --release --package xtask
	vtargetrun  "CARGO_MANIFEST_DIR=./xtask target/x86_64-unknown-linux-gnu/release/xtask generate-manpage"
}
```

Right now the rust target triple for the host architecture is hard-coded, as there doesn't seem to be a generic `xbps-src` variable for that since `XBPS_RUST_TARGET` is for the target. Definitely not